src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -61,5 +61,10 @@
         this.computer_cruiser = new Ship("Computer_Cruiser", 3);
         this.computer_destroyer = new Ship("Computer_Destroyer", 2);
         this.computer_submarine = new Ship("Computer_Submarine", 2);
+
+        this.playerHits = new int[0];
+        this.playerMisses = new int[0];
```

Comment:
Just kidding, I read up a bit on Java arrays just now. Whatever size we choose to give a dynamic array, all of the elements will be initialized to values of zero. Since we know that the maximum number of hits = total number of squares occupied by ships, and that the maximum number of misses is 100 minus those ship squares, we can use those numbers as the sizes for the arrays. And  since Java does automatic garbage collection anyway, this might be easier than starting with size 0 and dynamically sizing up as we go along.